### PR TITLE
DataViews: set proper role for AddFilter's items

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/55083
Follow-up to https://github.com/WordPress/gutenberg/pull/56676#discussion_r1412006580

### CODE REVIEW

File: packages/edit-site/src/components/dataviews/add-filter.js [hunk 1]
```diff
@@ -81,8 +81,6 @@ export default function AddFilter( { fields, view, onChangeView } ) {
 						{ filter.elements.map( ( element ) => (
 							<DropdownMenuItem
 								key={ element.value }
-								role="menuitemradio"
```
Comment (oandregal) [inline packages/edit-site/src/components/dataviews/add-filter.js:84]: @andrewhayward Given the component is a `DropdownMenuItem`, do I need to set the role explicitly `role="menuitem"`?

Comment (oandregal) [reply packages/edit-site/src/components/dataviews/add-filter.js:84]: The e2e test are passing, which means they are able to pick up the `menuitem` role for this component.

Comment (andrewhayward) [reply packages/edit-site/src/components/dataviews/add-filter.js:84]: [If you dig all the way down](https://github.com/radix-ui/primitives/blob/c31c97274ff357aea99afe6c01c1c8c58b6356e0/packages/react/menu/src/Menu.tsx#L706), it looks like the default role is `menuitem`, so no, we don't need to provide it here. There's an argument for being explicit, but I don't think it's really necessary given the component name.

**Review by andrewhayward — APPROVED** [review]
Looks good to me 🚀

**Pull request merged**